xbmc/pvr/recordings/PVRRecording.cpp

```diff
@@ -481,3 +482,28 @@ int CPVRRecording::ClientID(void) const
 {
   return m_iClientId;
 }
+
+bool CPVRRecording::IsInProgress() const
+{
+  // Note: It is not enough to only check recording time and duration against 'now'.
+  //       This can only be a first cheap quick check. Only the state of the related
+  //       timer is a safe indicator that the backend actually is recording this.
+
+  const CDateTime now(CDateTime::GetUTCDateTime());
+  if (m_recordingTime <= now && m_recordingTime + m_duration >= now)
+  {
+    const std::vector<CFileItemPtr> timers(g_PVRTimers->GetActiveRecordings());
+    for (const auto& timerItem : timers)
+    {
+      const CPVRTimerInfoTagPtr timer(timerItem->GetPVRTimerInfoTag());
+      if (timer->m_iClientId == m_iClientId &&
+          timer->ChannelTag()->UniqueID() == m_iChannelUid &&
+          timer->StartAsUTC() <= m_recordingTime &&
+          timer->EndAsUTC() >= m_recordingTime + m_duration)
```

Comment:
Is there some kind of ID we can check?

It's possible that 1 channel has 2 or more recordings running at the same time.
One of them is actually recording and the other one not (disabled for example, lower prio, no tuner, ...), in this case both recordings are marked as "recording" and only one should.